core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -207,6 +217,9 @@ public void uncaughtException(Thread t, Throwable e) {
   // switch to a ConcurrentHashMap.
   private final Set<InternalSubchannel> subchannels = new HashSet<>(16, .75f);
 
+  // Must be accessed from syncContext
+  private final Collection<RealChannel.PendingCall<?, ?>> pendingCalls = new LinkedHashSet<>();
```

Comment:
This may be large. We don't want to keep it around after we are no longer queuing calls.

Reply:
Yeah, I was trying to clear it but I had NPE 0d9449cda2858319e3ff34ba0cca857956ff1332. Seems I need introduce a flag.

Reply:
After each remove(), if the set is now empty check the config selector to see if buffering is still occurring. If not, set it to null.

Reply:
Done.